src/PathUtils.cpp

```diff
@@ -199,14 +199,23 @@ int PU::QuoteExternalPath(const char * path, char * buffer, int buffersize) {
 
 int PU::CreateLocalDir(const TCHAR * local) {
 	int res = SHCreateDirectoryEx(NULL, local, NULL);

```

Comment:
As we're already using `shlwapi`, why not simply use [`PathIsDirectory`](https://msdn.microsoft.com/en-us/library/windows/desktop/bb762131(v=vs.85).aspx) and add this at the start?
```c
    if (PathIsDirectory(local))
        return 0;
```
Rest of the code can remain as-is.